rust_src/src/process.rs

```diff
@@ -556,6 +556,52 @@ pub fn internal_default_interrupt_process(
 #[rustfmt::skip]
 def_lisp_sym!(Qinternal_default_interrupt_process, "internal-default-interrupt-process");
 
+/// Kill process PROCESS. May be process or name of one.
+/// See function `interrupt-process' for more details on usage.
+#[lisp_fn(min = "0")]
+pub fn kill_process(process: LispObject, current_group: LispObject) -> LispObject {
+    unsafe {
+        process_send_signal(process, libc::SIGKILL, current_group, false);
+    }
+    process
+}
+
+/// Send QUIT signal to process PROCESS. May be process or name of one.
+/// See function `interrupt-process' for more details on usage.
+#[lisp_fn(min = "0")]
+pub fn quit_process(process: LispObject, current_group: LispObject) -> LispObject {
+    unsafe {
+        process_send_signal(process, libc::SIGQUIT, current_group, false);
+    }
+    process
+}
+
+/// Stop process PROCESS. May be process or name of one.
+/// See function `interrupt-process' for more details on usage.
+/// If PROCESS is a network or serial or pipe connection, inhibit handling
+/// of incoming traffic.
+#[lisp_fn(min = "0")]
+pub fn stop_process(process: LispObject, current_group: LispObject) -> LispObject {
+    let mut p_ref: LispProcessRef = get_process(process).into();
```

Comment:
Do we need a `nil` check here ? The c code also checks with `PROCESSP`.

Reply:
I think this doesn't work like the c implementation. This throws an error, but it shouldn't.

Reply:
`process_send_signal` at the end of the function will throw the error so I guess that doesn't matter.

Reply:
Changing when errors are thrown could break some lisp.

Reply:
Do this check in a `match` instead would mimic the original flow more closely.

```
match get_process(process).as_process() {
    Some(p_ref) => {
        let process_type = p_ref.ptype();
        if process_type.eq(Qnetwork) || process_type.eq(Qserial) || process_type.eq(Qpipe) {
        unsafe {
            delete_read_fd(p_ref.infd);
            p_ref.command = Qt;
        }
        process
    },
    None => {
        #[cfg(windows)]
        error!("No SIGTSTP support");
        #[cfg(not(windows))]
        unsafe {
            process_send_signal(process, libc::SIGTSTP, current_group, false);
        }
    
        process
    }
}

Reply:
I don't see the difference, but I like it more.